www/data_module/src/services/data/collection/dataquery.service.coffee

```diff
@@ -0,0 +1,77 @@
+class DataQuery extends Factory
+    constructor: ($http, $q, API) ->
+        return class DataQueryClass
+            constructor: (query = {}) ->
+                @query = query
+                @filters = {}
+                for fieldAndOperator, value of query
+                    if ['field', 'limit', 'offset', 'order'].indexOf(fieldAndOperator) < 0
+                        if ['on', 'true', 'yes'].indexOf(value) > -1 then value = true
+                        else if ['off', 'false', 'no'].indexOf(value) > -1 then value = false
+                        @filters[fieldAndOperator] = value
+
+            computeQuery: (array) ->
+                # 1. filtering
+                @filter(array)
+
+                # 2. sorting
+                order = @query?.order
+                @sort(array, order)
+
+                # 3. limit
+                limit = @query?.limit
+                @limit(array, limit)
+
+
+            isFiltered: (v) ->
+                cmp = false
+                for fieldAndOperator, value of @filters
+                    [field, operator] = fieldAndOperator.split('__')
+                    switch operator
+                        when 'ne' then cmp = v[field] != value
+                        when 'lt' then cmp = v[field] <  value
+                        when 'le' then cmp = v[field] <= value
+                        when 'gt' then cmp = v[field] >  value
+                        when 'ge' then cmp = v[field] >= value
+                        else cmp = v[field] == value or
+                            (angular.isArray(v[field]) and value in v[field]) or
+                            # private fields added by the data service
+                            v["_#{field}"] == value or
+                            (angular.isArray(v["_#{field}"]) and value in v["_#{field}"])
```

Comment:
@tardyp I found the problem exists here. The `restService` successfully fetched the `builds` of a `builder` but at here they are all be filtered out.

The problems is that, in the query filter, we have a field named `builderid` with its value to be `"2"`(string). But in the array of `BuildInstance`, the `builderid` field is `2`(Integer). The coffee script translate `==` into `===`, so `"2" != 2`. Thus for BuildInstance this function always returns false. This causes the problem that prevent the displaying of `builds`  in `builder` page.